### PR TITLE
RATIS-1201. Enhance FileStore steaming tests

### PR DESCRIPTION
**Pull request by amaliujia**
https://issues.apache.org/jira/browse/RATIS-1201

### CODE REVIEW

**Comment by amaliujia** [comment]
R: @szetszwo @runzhiwang

File: ratis-examples/src/test/java/org/apache/ratis/examples/filestore/FileStoreStreamingBaseTest.java [hunk 1]
```diff
@@ -72,9 +77,30 @@ public void testFileStoreStream() throws Exception {
 
     final CheckedSupplier<FileStoreClient, IOException> newClient =
         () -> new FileStoreClient(cluster.getGroup(), getProperties(), raftPeer);
-    // TODO: configurable buffer size
-    final int bufferSize = 10_000;
-    testSingleFile("foo", SizeInBytes.valueOf("2M"), bufferSize, newClient);
+
+    testSingleFile("foo", SizeInBytes.valueOf("2M"), 10_000, newClient);
+    testSingleFile("foo", SizeInBytes.valueOf("2M"), 1000, newClient);
+    testSingleFile("foo", SizeInBytes.valueOf("20M"), 100_000, newClient);
```
Comment (runzhiwang) [inline ratis-examples/src/test/java/org/apache/ratis/examples/filestore/FileStoreStreamingBaseTest.java:83]: write the same file three times ?

Comment (amaliujia) [reply ratis-examples/src/test/java/org/apache/ratis/examples/filestore/FileStoreStreamingBaseTest.java]: My impression was the RandomAccessFile will overwrite existing file. 

After some research, it seems that when there is an existing file, RandomAccessFile only overwrites for the top of the file and leaves the rest of it.

So I have updated to use different file names to make the test more robust.

File: ratis-examples/src/test/java/org/apache/ratis/examples/filestore/FileStoreStreamingBaseTest.java [hunk 2]
```diff
@@ -84,36 +110,62 @@ private void testSingleFile(
...
+          () -> new StreamWriter().write(path, fileLength, bufferSize, newClient),
+          () -> path);
+      writerFutures.add(executor.submit(callable));
+    }
+    for (Future<StreamWriter> future : writerFutures) {
+      future.get();
+    }
+  }
+
+  static class StreamWriter {
```
Comment (szetszwo) [inline ratis-examples/src/test/java/org/apache/ratis/examples/filestore/FileStoreStreamingBaseTest.java:133]: Let's combine the StreamWriter to FileStoreBaseTest.Writer?  Then we can easily compare the performance in the unit tests.

**Review by szetszwo — COMMENTED** [review]
The change looks good.  Just a suggestion that how about combining the StreamWriter to FileStoreBaseTest.Writer, i.e. move StreamWriter.write(..) to FileStoreBaseTest.Writer.stream()?  Then, we can run the tests for write(), writeAsync() and stream() with the same parameters easily for comparing performance.

File: ratis-examples/src/test/java/org/apache/ratis/examples/filestore/FileStoreStreamingBaseTest.java (hunk 2 above)
Comment (amaliujia) [reply ratis-examples/src/test/java/org/apache/ratis/examples/filestore/FileStoreStreamingBaseTest.java:133]: Good point! Will refactor this code.

Comment (amaliujia) [reply ratis-examples/src/test/java/org/apache/ratis/examples/filestore/FileStoreStreamingBaseTest.java:137]: @szetszwo  Have combine writers into a single class.

**Review by szetszwo — APPROVED** [review]
+1 the change looks good.

**Pull request merged**